tools/fast/bin/localnet/main.go

```diff
@@ -0,0 +1,350 @@
+	// ImportAndStore
+	// 7. Generated some random data and import it to genesis
+	// 8. Genesis propposes a storage deal with miner
+	//
+	// WaitForDealState
+	// 9. Query deal till posted
+
+	var deals []*storagedeal.Response
+
+	for _, miner := range miners {
+		err = series.InitAndStart(ctx, miner)
+		if err != nil {
+			exitcode = handleError(err, "failed series.InitAndStart;")
+			return
+		}
+
+		err = series.Connect(ctx, genesis, miner)
+		if err != nil {
+			exitcode = handleError(err, "failed series.Connect;")
+			return
+		}
+
+		err = series.SendFilecoinDefaults(ctx, genesis, miner, fil)
+		if err != nil {
+			exitcode = handleError(err, "failed series.SendFilecoinDefaults;")
+			return
+		}
+
+		pledge := uint64(10)                    // sectors
+		collateral := big.NewInt(500)           // FIL
+		price := big.NewFloat(0.000000001)      // price per byte/block
+		expiry := big.NewInt(24 * 60 * 60 / 30) // ~24 hours
+
+		ask, err := series.CreateMinerWithAsk(ctx, miner, pledge, collateral, price, expiry)
+		if err != nil {
+			exitcode = handleError(err, "failed series.CreateMinerWithAsk;")
+			return
+		}
+
+		var data bytes.Buffer
+		dataReader := io.LimitReader(rand.Reader, int64(sectorSize))
+		dataReader = io.TeeReader(dataReader, &data)
+		_, deal, err := series.ImportAndStore(ctx, genesis, ask, files.NewReaderFile(dataReader))
+		if err != nil {
+			exitcode = handleError(err, "failed series.ImportAndStore;")
+			return
+		}
+
+		deals = append(deals, deal)
+
+	}
+
+	for _, deal := range deals {
+		err = series.WaitForDealState(ctx, genesis, deal, storagedeal.Posted)
+		if err != nil {
+			exitcode = handleError(err, "failed series.WaitForDealState;")
+			return
+		}
+	}
+
+	if shell {
+		client, err := env.NewProcess(ctx, lpfc.PluginName, options, fastenvOpts)
+		if err != nil {
+			exitcode = handleError(err, "failed to create client process;")
+			return
+		}
+
+		err = series.InitAndStart(ctx, client)
+		if err != nil {
+			exitcode = handleError(err, "failed series.InitAndStart;")
+			return
+		}
+
+		err = series.Connect(ctx, genesis, client)
+		if err != nil {
+			exitcode = handleError(err, "failed series.Connect;")
+			return
+		}
+
+		err = series.SendFilecoinDefaults(ctx, genesis, client, fil)
+		if err != nil {
+			exitcode = handleError(err, "failed series.SendFilecoinDefaults;")
+			return
+		}
+
+		interval, err := client.StartLogCapture()
+		if err != nil {
+			exitcode = handleError(err, "failed to start log capture;")
+			return
+		}
+
+		if err := client.Shell(); err != nil {
+			exitcode = handleError(err, "failed to run client shell;")
+			return
+		}
+
+		interval.Stop()
+		fmt.Println("===================================")
+		fmt.Println("===================================")
+		io.Copy(os.Stdout, interval) // nolint: errcheck
+		fmt.Println("===================================")
+		fmt.Println("===================================")
+	}
+
+	fmt.Println("Finished!")
+	fmt.Println("Ctrl-C to handleError")
```

Comment:
unclear to me whether and why i would want to handlerror...

Reply:
find and replace error; `s/exit/handleError/`